server/src/main/java/io/spine/server/model/ClassMap.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2018, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.model;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import io.spine.core.CommandClass;
+import io.spine.server.command.CommandHandlingClass;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import static com.google.common.collect.Sets.intersection;
+
+/**
+ * Maps a Java class to its {@link ModelClass}.
+ *
+ * @author Alexander Yevsyukov
+ */
+final class ClassMap {
+
+    private final Map<String, ModelClass<?>> classes = Maps.newConcurrentMap();
+
+    /**
+     * Returns a unique identifying name of the given class.
+     *
+     * <p>The returned value is guaranteed to be unique per class and non-null.
+     *
+     * @param cls the {@link Class} to identity
+     * @return a non-null class name
+     */
+    private static String nameOf(Class<?> cls) {
+        return cls.getName();
```

Comment:
@dmdashenkov, did you mean to do something more eventually in this method? The Javadoc hits on a bigger future for this method.

Reply:
The method is used to extract the `Class` key in a unified way for all the occurrences. We need this to avoid programming errors when using `Class.getName()` VS `Class.getSimpleName()` VS `Class.getCanonicalName()`.